doc/d-practical-exercises/src/robot_hand.py

```diff
@@ -44,19 +44,18 @@ def collision(self, c2, data=None, oMj1=None, oMj2=None):
             oMj2 = data.oMi[c2.jointParent]
         M1 = oMj1 * self.placement
         M2 = oMj2 * c2.placement
-
-        assert(self.isCapsule() and c2.isCapsule())
+        assert (self.isCapsule() and c2.isCapsule())
         l1 = self.length
         r1 = self.radius
         l2 = c2.length
         r2 = c2.radius
 
-        a1 = M1.act(np.matrix([0, 0, -l1 / 2]).T)
-        b1 = M2.act(np.matrix([0, 0, -l2 / 2]).T)
-        a2 = M1.act(np.matrix([0, 0, +l1 / 2]).T)
-        b2 = M2.act(np.matrix([0, 0, +l2 / 2]).T)
-
-        ab = pinv(np.hstack([a1 - a2, b2 - b1])) * (b2 - a2)
+        a1 = M1.act(np.array([0, 0, -l1 / 2]))
+        b1 = M2.act(np.array([0, 0, -l2 / 2]))
+        a2 = M1.act(np.array([0, 0, +l1 / 2]))
+        b2 = M2.act(np.array([0, 0, +l2 / 2]))
+        
+        ab = pinv((np.hstack((a1 - a2, b2 - b1))).reshape(3, 2)) * (b2 - a2)
```

Comment:
Personally, I cannot read this line. Too many parenthesis.
Either
```
        ab = pinv( np.hstack( [a1 - a2, b2 - b1] ).reshape(3, 2) ) * (b2 - a2)
```
or
```
        ab = pinv( np.array( [ a1 - a2, b1 - b2 ]).T) ) * (b2 - a2)
```
or split it into multiple lines with intermediate variable(s).

Reply:
I guess, it's done

Reply:
Thanks. Much clearer.